src/main/java/scala_maven/ScalaConsoleMojo.java

```diff
@@ -55,50 +75,263 @@
+        }
+
+        return classpath;
+    }
+
+    /**
+     * Construct the Classpath for any additional dependencies that are needed
+     * to run the Scala console.
+     * <p>
+     * This should include the following entities.
+     * <ul>
+     *   <li> Jline, used for readline like features in the REPL.
+     * </ul>
+     *
+     * @param scalaVersion the version of the Scala Compiler/Library we are
+     *        using for this execution.
+     *
+     * @return A {@link Set} of {@link String} of the classpath as defined by
+     *
+     * @throws {@link Exception} for many reasons, mostly relating to ad-hoc
+     *         dependency resolution.
+     */
+    private Set<String> setupConsoleClasspaths(final VersionNumber scalaVersion) throws Exception {
+        final Set<String> classpath =
+            new HashSet<String>();
+
+        super.addToClasspath(this.resolveJLine(scalaVersion,
+                                               this.fallbackJLine(scalaVersion)),
+                             classpath,
+                             true);
+
+        return classpath;
+    }
+
+    /**
+     * Attempt to resolve JLine against the Scala Compiler's dependency tree.
+     * <p>
+     * This allows us to not have to worry about manually keeping the JLine
+     * dependency synchronized with the Scala REPL, which is likely to cause
+     * binary compatibility errors. If, for some reason, we are unable to
+     * resolve this dependency using the Compiler's dependency tree, we fallback
+     * to a set of hard-coded defaults that will <em>usually</em> work, but will
+     * not always work.
+     * <p>
+     * If the dynamic approach to finding the JLine dependency proves stable, we
+     * may drop the fallback in the future.
+     *
+     * @param scalaVersion the version of the Scala Compiler/Library we are
+     *        using for this execution.
+     * @param defaultFallback returned if we are unable to resolve JLine against
+     *        the Scala Compiler's dependency tree.
+     *
+     * @return an {@link Artifact} to provide to the runtime of the Scala
+     *         console conforming the JLine.
+     *
+     * @throws {@link Exception} for many reasons, mostly relating to ad-hoc
+     *         dependency resolution.
+     */
+    private Artifact resolveJLine(final VersionNumber scalaVersion,
+                                  final Artifact defaultFallback) throws Exception {
+        final Artifact compilerArtifact =
+            super.scalaCompilerArtifact(scalaVersion.toString());
+        final Set<Artifact> compilerDeps =
+            super.resolveArtifactDependencies(compilerArtifact);
+        for (final Artifact a : compilerDeps) {
+            super.getLog().warn(a.toString());
+            if (this.filterForJline(a)) {
+                return a;
             } else {
-                jcmd.addJvmArgs("-noverify", "-javaagent:" + javaRebelPath.getCanonicalPath());
+                continue;
```

Comment:
Will do. It is a personal preference of mine to have fully expanded branches, but I recognize that is a subjective point of view.